src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -360,6 +360,60 @@ public static ImmutableArray<ParameterSymbol> DelegateParameters(this TypeSymbol
             return type.DelegateInvokeMethod().Parameters;
         }
 
+        public static bool TryGetElementTypesIfTupleOrCompatible(this TypeSymbol type, out ImmutableArray<TypeSymbol> elementTypes)
+        {
+            if (type.IsTupleType)
+            {
+                elementTypes = ((TupleTypeSymbol)type).TupleElementTypes;
+                return true;
+            }
+
+            // The following codepath should be very uncommon since it would be rare
+            // to see a tuple underlying type not represented as a tuple.
+            // It still might happen since tuple underlying types are creatable via public APIs 
+            // and it is also possible that they would be passed in.
+
+            // PERF: if allocations here become nuisance, consider caching the results
+            //       in the type symbols that can actually be tuple compatible
+            int cardinality;
+            if (!type.IsTupleCompatible(out cardinality))
```

Comment:
My understanding is that for tuple conversions that are elementwise (not identity ones), the underlying type is irrelevant since we would be unpacking/repacking anyways.
We can clarify that, but that is my understanding 

Example: you cannot pass _byref_ a tuple argument to a tuple parameter of the same structural signature if there is mismatch in underlying types. However you can pass _byval_ in a similar situation and a copy will be performed. 

I definitely asked that question when we discussed that, and the answer was "yes".

Basically conversion from (int, int)^ to (int, int)^^ of different underlying type is classified as ImplicitTuple conversion. We cannot identity-convert, but we can convert elementwise.

Indeed, this method's purpose is basically to strip away the underlying type and give back the types of elements.